modules/product/product.service.js

```diff
@@ -64,13 +63,32 @@ class ProductsService {
         },
       ];
     }
-
     return Products.find(filters)
       .skip(skip)
       .limit(limit)
       .sort(sort);
   }
 
+  getProductsCount({ filter, search }) {
+    const isNotBlank = str => !(!str || str.trim().length === 0);
+    const filters = this.filterItems(filter);
+
+    if (isNotBlank(search)) {
```

Comment:
You duplicate generating filters in this method and `getProducts`, please, move it to separate method, and call it from to places

Reply:
fixed